src/worldcereal/rdm_api/rdm_interaction.py

```diff
@@ -219,50 +362,100 @@ def _setup_sql_query(
 
         return combined_query
 
-    def query_rdm(
+    def download_samples(
         self,
-        geometry: BaseGeometry,
-        temporal_extent: Optional[List[str]] = None,
+        collection_ids: Optional[List[str]] = None,
         columns: List[str] = DEFAULT_COLUMNS,
+        geometry: Optional[BaseGeometry] = None,
+        temporal_extent: Optional[List[str]] = None,
+        ewoc_codes: Optional[List[int]] = None,
+        include_public: Optional[bool] = True,
+        include_private: Optional[bool] = False,
     ):
-        """Queries the RDM API and generates a GeoParquet file of all intersecting sample IDs.
+        """Queries the RDM API and generates a Pandas GeoDataframe of all samples meeting the search criteria.
 
         Parameters
         ----------
-        geometry : BaseGeometry
-            A user-defined polygon. CRS should be EPSG:4326.
-        temporal_extent : List[str], optional
-            A list of two strings representing the temporal extent, by default None. If None, all available data will be queried.
-            Dates should be in the format "YYYY-MM-DD".
+        collection_ids : Optional(List[str]), optional
+            List of collection IDs to download samples from.
+            If not specified, all collections matching the search criteria defined by
+            the other input parameters will be queried.
         columns : List[str], optional
             A list of column names to extract., by default DEFAULT_COLUMNS
+        geometry : Optional[BaseGeometry], optional
+            A user-defined geometry for which all intersecting collections need to be found.
+            CRS should be EPSG:4326.
+            If None, all available data will be queried., by default None
+        temporal_extent : List[str], optional
+            A list of two strings representing the temporal extent, by default None.
+            If None, all available data will be queried.
+            Dates should be in the format "YYYY-MM-DD".
+        ewoc_codes: Optional[List[int]] = None
+            If specified, only samples with the specified EWOC codes will be extracted.
+        include_public: Optional[bool] = True
+            Whether or not to include public collections.
+        include_private: Optional[bool] = False
+            Whether or not to include private collections.
 
         Returns
         -------
         gpd.GeoDataFrame
             A GeoDataFrame containing the extracted columns and the geometry.
         """
-        collection_ids = self._collections_from_rdm(
-            geometry=geometry, temporal_extent=temporal_extent
-        )
+
+        # Determine which collections need to be queried if they are not specified
+        if not collection_ids:
```

Comment:
similar comment as earlier. Should either enforce this or always make the query (maybe the latter is better).

Reply:
with "similar comment as before", you mean I should change this to "if collection_ids is None: ?